src/compiler/crystal/semantic/call_error.cr

```diff
@@ -136,7 +136,7 @@ class Crystal::Call
 
     # If we made a lookup without the special rule for literals,
     # and we have literals in the call, try again with that special rule.
-    if with_literals == false && (args.any? { |arg| arg.is_a?(NumberLiteral) || arg.is_a?(SymbolLiteral) } ||
```

Comment:
FWIW `with_literals` is not restricted to bool, although it is used as such. So although I am fine with this change, it is not necessarily a sound change in general.